File: src/NeZoviReg.Domain/Model/RegUser.cs
using System.Runtime.CompilerServices;
using NeZoviReg.Domain.Auth;

namespace NeZoviReg.Domain.Model;

/// <summary>
/// Users of the 'Ne_zovi' registry.
/// </summary>
public class RegUser: Entity
{
    public RegUser(string firstName, string lastName, string email)
    {
        FirstName = firstName;
        LastName = lastName;
        Email = email;
    }

    public RegUser(int id, string firstName, string lastName, string email)
               :base(id)
    {
        FirstName = firstName;
        LastName = lastName;
        Email = email;
    }

    public string FirstName { get; private set; }

    public string LastName { get; private set; }

    public string FullName => $"{FirstName} {LastName}";

    public string Email { get; private set; }

    public string ThumbPrint { get; private set; } = string.Empty;

    private readonly List<Role> _roles = new();
    public IReadOnlyCollection<Role> Roles => _roles;



}

File: angular/src/app/domain/model/regUser.ts
import { Entity } from "./entity";
import { RoleType } from "./roleType";

export interface RegUser extends Entity {
    guidId: string;
    companyName: string;
    firstName: string;
    lastName: string;
    address: string;
    taxNumber: string;
    regNumber: string;
    userName: string;
    password: string;
    email: string;
    roles: RoleType[] | null | undefined;
}